modules/rest-api/pages/system-secrets-configuration.adoc

```diff
@@ -0,0 +1,316 @@
+= Configuring System Secrets
+:description: By means of the REST API, System Secrets can be configured, and the current configuration can be retrieved.
+
+[abstract]
+{description}
+
+[#http-methods-and-uris]
+== HTTP Methods and URIs
+
+----
+GET /nodes/self/secretsManagement
+
+POST /node/controller/secretsManagement
+----
+
+[#description]
+== Description
+Configures the _encryption service_; which comprises the master password, encryption keys, key storage, and the location of key-control scripts.
+
+The *Full Admin*, *Local User Security Admin*, or *External User Security Admin* role is required.
+
+== Curl Syntax
+
+----
+curl -X GET http://<ip-address-or-hostname>:8091/nodes/self/secretsManagement
+  -u <username>:<password>
+
+curl -X POST http://<ip-address-or-hostname>:8091/node/controller/secretsManagement
+  -u <username>:<password>
+  -H "Content-Type:application/json"
+  -d "$(cat)" | jq <payload>
+----
+
+The `payload` must be a JSON document of the following form:
+
+----
+{
+  "encryptionService": {
+    [<option>: <value>]*
+  }
+}
+----
+
+Each `option` is described below.
+
+* `keyEncrypted`.
+Whether encryption keys should be encrypted by means of the master password.
+The value can be either `true` (the default) or `false`.
+
+* `keyPath`.
+Specifies whether the encryption keys are stored at the default or at a custom location.
+The value can be  either `auto` (which is the default) or `custom`.
+If the value is `auto`, the default location is used.
+If the value is `custom`, the path provided by `customKeyPath` is used.
+Note that this option is used only if `keyStorageType` has the value `file`.
+
+* `customKeyPath`.
+A file-path that specifies a custom location at which the encryption keys are stored.
+This file-path is used only if the value of `keyPath` is `custom`.
+
+* `keyStorageType`.
+The value can be `file` (which is the default) or `script`.
+If the value is `file`, the system stores the encryption keys in a file.
+If the value is `script`, a user-specified script is called, when keys are to be written, read, or deleted.
+These scripts must be specified by `writeCmd`, `readCmd`, and `deleteCmd`, respectively.
+
+* `passwordSource`.
+The value can be `env` (which is the default) or `script`.
+If the value is `env`, the master password is read from the environment variable specified by `passwordEnv`.
+If the value is `script`, the master password is provided by the script specified by `passwordCmd`.
+
+* `passwordEnv`.
+The name of the environment variable from which the master password is read (provided that the value of `passwordSource` is `env`).
+By default, the value of `passwordEnv` is `CB_MASTER_PASSWORD`.
+
+* `passwordCmd`.
```

Comment:
We need to describe how this command is used:
1. When it is executed: normally when node is starting, but also can be called if secrets manager restarts which can be caused by reconfiguration or by a bug.
2. It is expected that this command exits with status code 0. If exit status is different it is considered as an error.
3. It is expected that this command writes the master password to stdout.

Reply:
I've added.